models/userStatus.js

```diff
@@ -193,4 +205,133 @@ const updateAllUserStatus = async () => {
   }
 };
 
-module.exports = { deleteUserStatus, getUserStatus, getAllUserStatus, updateUserStatus, updateAllUserStatus };
+/**
+ * Updates the user status based on a new task assignment.
+ * @param {string} userId - The ID of the user.
+ * @returns {Promise<{
+ *   status: string,
+ *   message: string,
+ *   data?: object
+ * }>} - The response object indicating the status of the user status update.
+ * @throws {Error} If there is an error retrieving or updating the user status.
+ */
+const updateUserStatusOnNewTaskAssignment = async (userId) => {
+  try {
+    let latestStatusData;
+    try {
+      latestStatusData = await getUserStatus(userId);
+    } catch (error) {
+      logger.error("Unable to retrieve the current status" + error.message);
+      throw new Error("Unable to retrieve the current status");
+    }
+    const { userStatusExists } = latestStatusData;
+    if (!userStatusExists) {
+      return createUserStatusWithState(userId, userStatusModel, userState.ACTIVE);
+    }
+    const {
+      data: {
+        currentStatus: { state },
+      },
+    } = latestStatusData;
+    if (state === userState.ACTIVE) {
+      return generateAlreadyExistingStatusResponse(userState.ACTIVE);
+    } else if (state === userState.IDLE || state === userState.ONBOARDING) {
```

Comment:
```suggestion
    if (state === userState.ACTIVE) {
      return generateAlreadyExistingStatusResponse(userState.ACTIVE);
    } if (state === userState.IDLE || state === userState.ONBOARDING) {
```

Reply:
yes they are mutually exclusive and that will be maintained as you have an early return if one of them fulfills we will automatically not go to the other

Reply:
Done